### PR TITLE
Editor/Navigator: Fix - Resizeable not working on the first time (PT#587)

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: assets/dev/js/editor/regions/navigator/navigator.js [hunk 1]
```diff
@@ -92,7 +92,9 @@ export default class extends BaseRegion {
 
 		this.$el.draggable( this.getDraggableOptions() );
 
-		this.$el.resizable( this.getResizableOptions() );
+		setTimeout( () => {
```
Comment (matipojo) [inline assets/dev/js/editor/regions/navigator/navigator.js]: Why it happen? and what about `this.$el.draggable`?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/regions/navigator/navigator.js (hunk 1 above)
Comment (matipojo) [reply assets/dev/js/editor/regions/navigator/navigator.js]: @iNewLegend ?

Comment (iNewLegend) [reply assets/dev/js/editor/regions/navigator/navigator.js]: bug in JQUERY UI as i understand.

Comment (matipojo) [reply assets/dev/js/editor/regions/navigator/navigator.js]: What about  `draggable`?

Comment (iNewLegend) [reply assets/dev/js/editor/regions/navigator/navigator.js]: the conflict is between the both.

Comment (matipojo) [reply assets/dev/js/editor/regions/navigator/navigator.js]: Add an explanation.

**Review by matipojo — APPROVED** [review]

File: assets/dev/js/editor/regions/navigator/navigator.js [hunk 2]
```diff
@@ -92,7 +92,11 @@ export default class extends BaseRegion {
 
 		this.$el.draggable( this.getDraggableOptions() );
 
-		this.$el.resizable( this.getResizableOptions() );
+		// Used timeout here since draggable and resizable are conflicting.
```
Comment (kobizz) [inline assets/dev/js/editor/regions/navigator/navigator.js]: This issue didn't exist in the past. It's probably due to another change that screwed it up

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/regions/navigator/navigator.js (hunk 2 above)
Comment (matipojo) [reply assets/dev/js/editor/regions/navigator/navigator.js]: @iNewLegend , try chaining, like here:
https://stackoverflow.com/a/16990819/5503355

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/regions/navigator/navigator.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/regions/navigator/navigator.js]: > This issue didn't exist in the past. It's probably due to another change that screwed it up
Yes ( i tested in 2.7.5 ).


> @iNewLegend , try chaining, like here:
> https://stackoverflow.com/a/16990819/5503355

tried didn't resolve the issue.

**Review by matipojo — DISMISSED** [review]

File: assets/dev/js/editor/regions/navigator/navigator.js (hunk 2 above)
Comment (kobizz) [reply assets/dev/js/editor/regions/navigator/navigator.js]: Nope. This issue for sure didn't exist before.

Comment (iNewLegend) [reply assets/dev/js/editor/regions/navigator/navigator.js]: The issue was caused beacuse of `$e.route( 'navigator' )`, i don't know why but when `initLayout` was called via `$e.route` resize able become screwed.

tested on document change resize is working.
https://github.com/elementor/elementor/pull/10854/files#diff-2941c4dbb172f73b42807fa33f8fc742R294

**Comment by iNewLegend** [comment]
@matipojo, @kobizz 

So there was two issues:
1. Navigator was rendered twice.
2. When really it needed to be re-rendered there was no destroy for 'ui-resizable'.

PS: seems to be required to destroy and create again after re-render of layout

File: assets/dev/js/editor/regions/navigator/navigator.js [hunk 3]
```diff
@@ -87,18 +88,35 @@ export default class extends BaseRegion {
 		};
 	}
 
+	onBeforeShow() {
```
Comment (matipojo) [inline assets/dev/js/editor/regions/navigator/navigator.js]: Create a destroy method and call it on close. no explanation needed.

File: assets/dev/js/editor/regions/navigator/navigator.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ export default class extends BaseRegion {
 		this.component = $e.components.register( new Component( { manager: this } ) );
 
 		this.isDocked = false;
+		this.isLayoutInitOnce = false;
```
Comment (matipojo) [inline assets/dev/js/editor/regions/navigator/navigator.js]:
```suggestion
		this.hasLayout= false;
```

File: assets/dev/js/editor/regions/navigator/navigator.js [hunk 5]
```diff
@@ -87,18 +88,35 @@ export default class extends BaseRegion {
...
 
 	open( model ) {
 		// If open once.
 		if ( ! this.opened ) {
-			this.initLayout();
+			/**
+			 *  Since navigator is static element its layout should be rendered once.
+			 *  reason: two flows opens the navigator: $e.route( 'navigator' ) and on documentLoaded.
+			 */
+			if ( ! this.isLayoutInitOnce ) {
```
Comment (matipojo) [inline assets/dev/js/editor/regions/navigator/navigator.js]: you can remove the comment, `hasLayout` is enough.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/regions/navigator/navigator.js [hunk 6]
```diff
@@ -91,15 +91,12 @@ export default class extends BaseRegion {
...
 		this.$el.resizable( this.getResizableOptions() );
 	}
 
 	open( model ) {
 		// If open once.
 		if ( ! this.opened ) {
-			this.initLayout();
-
 			this.opened = true;
 		}
```
Comment (kobizz) [inline assets/dev/js/editor/regions/navigator/navigator.js]: These lines are now redundant

File: assets/dev/js/editor/regions/navigator/navigator.js [hunk 7]
```diff
@@ -135,6 +132,10 @@ export default class extends BaseRegion {
 			this.saveStorage( 'visible', false );
 		}
 
+		if ( this.$el.hasClass( 'ui-resizable' ) ) {
```
Comment (kobizz) [inline assets/dev/js/editor/regions/navigator/navigator.js]: The official way to check it is by calling `this.$el.resizable( 'instance' )`

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/regions/navigator/navigator.js (hunk 6 above)
Comment (kobizz) [reply assets/dev/js/editor/regions/navigator/navigator.js:101]: Why did you resolve this comment?

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/regions/navigator/navigator.js (hunk 6 above)
Comment (iNewLegend) [reply assets/dev/js/editor/regions/navigator/navigator.js:101]: redundant means they not needed anymore?

Comment (iNewLegend) [reply assets/dev/js/editor/regions/navigator/navigator.js:101]: @kobizz please explain what what you mean or what you want me to do

Comment (kobizz) [reply assets/dev/js/editor/regions/navigator/navigator.js:101]: The whole block of code should be removed:
```
// If open once.
if ( ! this.opened ) {
	this.opened = true;
}

**Review by kobizz — APPROVED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**